### PR TITLE
Automatically calculate milestone expected ages

### LINKED ISSUE #133: Automatically calculate Milestone.expected_age_months
The value of `Milestone.expected_age_months` should be set based on the averaged results for that milestone, according to some to-be-determined scheme, e.g.

"From a histogram of scores vs age, binned in 3 month intervals, the lowest age where the mean score is above 3"

Every so often, the server should go throught the data & recalculate these values.

It may be good to add a human approval step before applying the changes to ensure the data look reasonable.

There could also be a filter to filter out obviously bogus entries, once we have sufficient data to identify this.

### LINKED ISSUE #153: Visualisation plots/charts
For the admin/research interace it would be nice to offer some basic visualisation of the data.

In particular for #133 it would be nice to show the old & new expected ages as lines on a plot of score vs age data points.

[unovis](https://unovis.dev/) looks like a nice option, which comes with a svelte version although they [don't yet support svelte 5](https://github.com/f5/unovis/issues/379)

### PR DESCRIPTION
**Pull request by lkeegan**
- uses placeholder algorithm for now
  - calculates average score for a milestone from each age (1 month bins)
  - expected age is the first age with an average score >= 3
  - the details of this algorithm can be refined later
  - resolves #133
- add view data button below expected age slider in admin edit milestone dialog
  - displays a plot of average scores vs age for this milestone
  - includes a dotted line showing the calculated expected age based on the data
- add admin tab for this
  - displays expected age for each milestone & button to plot data
  - button to calculate new expected age for each
  - button to save changes
- add @unovis/ts chart/plotting library
  - using @unovis/ts for now due to lack of svelte 5 support in @unovis/svelte
  - can switch to the svelte version once this issue is resolved
  - svelte-vega looked more promising but gave an error at build time
  - resolves #153
- add numpy to backend dependencies
- remove default constructors from some models to improve generated typescript types
  - e.g. `milestones: list[MilestoneAdmin] = {}` used to give typescript type `MilestoneAdmin[] | undefined`
  - replace with `milestones: list[MilestoneAdmin]` to give `MilestoneAdmin[]`

### CODE REVIEW

**Review by MaHaWo — APPROVED** [review]
some frontend changes confused me a little bit. otherwise looks good.

File: frontend/src/lib/components/Admin/MilestoneExpectedAges.svelte [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+let currentMilestoneId = $state(null as number | null);
+let showMilestoneExpectedAgeModal = $state(false);
+let currentTitle = $state("");
+let expectedAges = $state({} as Record<number, MilestoneAgeScores>);
+let calculateProgress = $state(0);
+let saveProgress = $state(0);
+
+async function getNewExpectedAge(milestoneId: number) {
+	const { data, error } = await getMilestoneAgeScores({
```
[the remark this answers is not in the document]
Comment (lkeegan) [reply frontend/src/lib/components/Admin/MilestoneExpectedAges.svelte:36]: this endpoint is for a single milestone: it returns the expected age + the data for plotting purposes:
```ts
export type MilestoneAgeScores = {
    scores: Array<MilestoneAgeScore>;
    expected_age: number;
};
```
where
```ts
export type MilestoneAgeScore = {
    age_months: number;
    avg_score: number;
    expected_score: number;
};
```

File: frontend/src/locales/de.json [hunk 2]
```diff
@@ -41,8 +41,14 @@
 		"help": "Förderhilfen",
 		"image": "Bild",
 		"images": "Bilder",
+		"age": "Alter",
+		"average-score": "Durchschnittsnote",
```
[the remark this answers is not in the document]
Comment (lkeegan) [reply frontend/src/locales/de.json:45]: Better translation very welcome :-)
(the average score here refers to the average of the (1-4) answers given for this milestone)

[the remark this answers is not in the document]
Comment (MaHaWo) [reply frontend/src/locales/de.json:45]: it doesn´t make much of a difference, personally I would prefer "Durchschnittspunktzahl" (direct translation of 'average score'), and as far as I have seen more commonly used when one assesses some performance numerically while `Note` is used when wants to add a judgement (product tests, school, ...) , but that's admittedly a bit of  hair splitting here

[the remark this answers is not in the document]
Comment (lkeegan) [reply frontend/src/locales/de.json:45]: Durchschnittspunktzahl sounds good, thanks!

[the remark this answers is not in the document]
Comment (MaHaWo) [reply frontend/src/locales/de.json]: other variants: Durchschnittliche Einschätzung (average assessment), or just Durchschnittswert (average value).

**Pull request merged**